iothub/device/src/Common/Extensions/CommonExtensions.cs

```diff
@@ -61,15 +61,10 @@ public static string GetValueOrDefault(this IDictionary<string, string> map, str
                 throw new ArgumentException("Malformed Token");
             }
 
-            IEnumerable<string[]> parts = new Regex($"(?:^|{kvpDelimiter})([^{kvpDelimiter}{kvpSeparator}]*){kvpSeparator}")
-                .Matches(valuePairString)
-                .Cast<Match>()
-                .Select(m => new string[] {
-                    m.Result("$1"),
-                    valuePairString.Substring(
-                        m.Index + m.Value.Length,
-                        (m.NextMatch().Success ? m.NextMatch().Index : valuePairString.Length) - (m.Index + m.Value.Length))
-                });
+            IEnumerable<string[]> parts = valuePairString
+                .Split(kvpDelimiter)
+                .Where(p => p.Trim().Length > 0)
+                .Select(p => p.Split(new char[] { kvpSeparator }, 2));
```

Comment:
You can save off this character array as as `static readonly` so it is not allocated each time

Reply:
> le that there are registered devices on older IoT hubs that still have devices with semicolons in their names? If so, this would break those users

Good point. Any suggestion on how we can figure this out? Would checking Kusto service logs be sufficient? Or should I talk with someone on the Hub team?